runtime/gc_structs/ClassIterator.cpp

```diff
@@ -101,6 +101,14 @@ GC_ClassIterator::nextSlot()
 		}
 		_state += 1;
 
+	case classiteratorclassslots_state_valueType_defaultValue:
+		_state += 1;
```

Comment:
make an iterator that has a boolean state variable to record if the default value has been returned or not. Use that here instead

Reply:
@tajila I made an iterator of boolean type. Do I keep this case and place the if statement inside the case or I just replace this case with the if statement? then the if statement would become a part of the last case.